src/Microsoft.DotNet.Build.CloudTestTasks/UploadClient.cs

```diff
@@ -84,6 +84,9 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                     using (HttpClient client = new HttpClient())
                     {
                         client.DefaultRequestHeaders.Clear();
+
+                        // In random occassions the request fails if the network is slow and it takes more than 100 seconds to upload 4MB. 
+                        client.Timeout = TimeSpan.FromMinutes(5);
```

Comment:
Did we consider making the timeout an argument to the build task? Would amke it easer to configure in the future: change a build property vs. rebuilding the DLL.

Reply:
I did but thought this might not be a thing we'd like to change that often but I'll just add 5 as default and then the value can be overridden if needed